src/firebase/CRUDServices/selfImageExperimentService.js

```diff
@@ -0,0 +1,116 @@
+import firebase from '../firebase';
+
+export default class SelfImageExperimentService {
+  constructor() {
+    self.db = firebase.firestore();
+  }
+
+    // singleton instance.
+    static sieInstance = null;
+
+    /**
+     * creating singleton instance.
+     * @return {SelfImageExperimentService} instance
+     */
+    static getInstance = () => {
+      if (self.sieInstance == null) {
+        self.sieInstance = new SelfImageExperimentService();
+      }
+      return self.sieInstance;
+    }
+
+    /**
+     * Create self image experiment in database.
+     * @param {String} experimentId id of the new self image experiment.
+     * @param {SelfImageExperiment} selfImageExperiment
+     * self image experiment object.
+     * @return {Boolean} succeed or not.
+     */
+    postExperiment = async (experimentId, selfImageExperiment) => {
+      try {
+        await db.collection('Experiments').doc(experimentId)
+          .set(selfImageExperiment);
+        return true;
+      } catch (err) {
+        // TODO: add logic here
+        return false;
+      }
+    }
+
+    /**
+     * Get all existing self image experiments.
+     * @return {SelfImageExperiment[]} array of self image experiment object.
+     */
+    getExperiments = async () => {
+      try {
+        const experimentRef = db.collection('SelfImageExperiments');
+        const snapshot = await experimentRef.get();
+        if (snapshot.empty) {
+          return;
+        }
+
+        snapshot.forEach( (doc) => {
+          // TODO: add logic here
+        });
+      } catch (err) {
+        // TODO: add logic here
+      }
```

Comment:
shouldn't this return SelfImageExperiment[] per documentation? Both try and catch blocks are missing return statements